presto-hive/src/main/java/io/prestosql/plugin/hive/RecordFileWriter.java

```diff
@@ -120,8 +122,13 @@ public long getWrittenBytes()
         }
 
         if (committed) {
+            if (finalWrittenBytes != -1) {
+                return finalWrittenBytes;
+            }
+
             try {
-                return path.getFileSystem(conf).getFileStatus(path).getLen();
+                finalWrittenBytes = path.getFileSystem(conf).getFileStatus(path).getLen();
```

Comment:
@electrum : we very recently faced an issue where a query failed in this file status call (FNFE exception). the table is on s3 - seems like an EC issue since it does not happen consistently. should we consider adding retry as well here before failing the query?

Reply:
We could retry here, but this gets messy. It might be better to try and implement `ExtendedRecordWriter` for all of the formats we support -- which we need anyway for writer scaling to work.

Reply:
Yes, it will get messy. `getWrittenBytes` is called at other places as well right now - like for writer scaling and for partition write stats. a complete solution for fixing extra call and s3 EC would be implementing `ExtendedRecordWriter`.